common/addons/library-image/src/main/java/com/dfsek/terra/addons/image/config/ImageLibraryPackConfigTemplate.java

```diff
@@ -0,0 +1,39 @@
+package com.dfsek.terra.addons.image.config;
+
+import com.dfsek.tectonic.api.config.template.ConfigTemplate;
+import com.dfsek.tectonic.api.config.template.annotations.Default;
+import com.dfsek.tectonic.api.config.template.annotations.Description;
+import com.dfsek.tectonic.api.config.template.annotations.Value;
+
+import com.dfsek.terra.api.properties.Properties;
+
+
+public class ImageLibraryPackConfigTemplate implements ConfigTemplate, Properties {
+    // TODO - These would be better as plugin wide config parameters in config.yml
+    
+    @Value("images.cache.load-on-use")
+    @Description("If set to true, images will load into memory upon use rather than on pack load.")
```

Comment:
this is a question for @Astrashh to answer

Reply:
Some considerations that can be weighed up:

If this is enabled, it moves any potential IO exceptions related to loading the images to generation time rather than on pack load. Some exceptions should really be caught during pack load, I wanted to include a check for if the file exists but there wasn't a trivial way to do that with the way the current `Loader` impl works (dfsek has said in the past that this should probably be replaced with java nio). IIRC an exception thrown by the lazily loaded version causes the server to crash.

If the config pack uses very large images, there is a potential for large lag spikes during generation depending on image size and hardware. This isn't something I've really tested though. (My testing mostly just involved checking if the system actually works as expected in typical generation using a small basic image). This is not as much of an issue if a config pack splits a larger image into lots of smaller images to be stitched together.

I consider these performance options a more experimental, but changing the default to true in future wouldn't be a big issue.

Reply:
Also the options should only really be used in cases where the area the image has been used for has already been generated, e.g. in limited size pregenerated worlds, or images that are used for a spawn area.